src/modules/enrollment/components/pages/EnrollmentCurrentLivingSituationsPage.tsx

```diff
@@ -51,14 +52,23 @@ const EnrollmentCurrentLivingSituationsPage = () => {
   const enrollmentId = enrollment?.id;
   const clientId = enrollment?.client.id;
 
+  const clsFeature = useMemo(
+    () =>
+      enrollment?.dataCollectionFeatures.find(
+        (f) => f.role === DataCollectionFeatureRole.CurrentLivingSituation
+      ),
+    [enrollment?.dataCollectionFeatures]
+  );
+
   const evictCache = useCallback(() => {
     cache.evict({
       id: `Enrollment:${enrollmentId}`,
       fieldName: 'currentLivingSituations',
     });
   }, [enrollmentId]);
 
-  const canEditCls = enrollment?.access?.canEditEnrollments || false;
+  const canEditCls =
+    (enrollment?.access?.canEditEnrollments && !clsFeature?.legacy) || false;
 
   const localConstants = useMemo(
     () => ({
```

Comment:
What I'm not seeing, but was expecting to see, is some change to the logic for how FormDefinitions are fetched when viewing/editing these records. Tracing the code, it should always come back  to `useFormDefinition` hook (from ViewRecord or useFormDialog).

I haven't tested it locally yet, but I would expect an error if I attempt to open, say, a CurrentLivingSItuation if CLS doesn't have any form instances enabled for this project. That's the piece where we want to maybe choose the one that's [managed in version control (tbd)](https://github.com/greenriver/hmis-warehouse/pull/4869) but for now we can just pick the "first". I _think_ we need to make sure that change goes in together. This change may result in more historical data being displayed to users, and we don't want them to face errors when trying to view those records.

Reply:
(Hmm and that's probably a change to the backend w.r.t how form definitions are resolved- just commenting here since I was looking for it )

Reply:
Hm, yes I agree I need to address that and it should go into this PR. It's missing here because I had held off on it per our discussion last week (https://github.com/open-path/Green-River/issues/6159#issuecomment-2430234508), sorry I did not clarify that! Sounds it is not actually blocked by https://github.com/open-path/Green-River/issues/6763 and I should proceed with just picking the first?